stdlib/public/core/StringStorage.swift

```diff
@@ -121,26 +121,21 @@ extension _AbstractStringStorage {
 
     // Handle the case where both strings were bridged from Swift.
     // We can't use String.== because it doesn't match NSString semantics.
-    let knownOther = _KnownCocoaString(other)
-    switch knownOther {
+    switch CocoaStringKind(other) {
     case .storage:
       return _nativeIsEqual(
         _unsafeUncheckedDowncast(other, to: __StringStorage.self))
     case .shared:
       return _nativeIsEqual(
         _unsafeUncheckedDowncast(other, to: __SharedStringStorage.self))
-#if !(arch(i386) || arch(arm))
-    case .tagged:
-      fallthrough
-#endif
-    case .cocoa:
+    default:
```

Comment:
Eventually we may want to bring the tagged special case back if we add dedicated tagged pointer unpacking stuff to CF. Hopefully CFStringGetBytes can be fast enough for that case though.